perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_def_login_namespace_umbrellaid_persistent_shadow.java

```diff
@@ -31,26 +33,42 @@
 	private final static Logger log = LoggerFactory.getLogger(
 		urn_perun_user_attribute_def_def_login_namespace_umbrellaid_persistent_shadow.class);
 
+	private static final String A_U_UMBRELLAID_IDENTIFIER = AttributesManager.NS_USER_ATTR_DEF + ":umbrellaIDIdentifier";
 	private final static String CONFIG_EXT_SOURCE_NAME_UMBRELLA_ID = "extSourceNameUmbrellaID";
 	private final static String FRIENDLY_NAME = "login-namespace:umbrellaid-persistent-shadow";
 	private final static String FRIENDLY_NAME_PARAMETER = "umbrellaid-persistent-shadow";
 
 	private final ModulesConfigLoader loader = new ModulesYamlConfigLoader();
 
 	/**
-	 * fillAttribute will generate a version 1 UUID
+	 * fillAttribute will set umbrellaIDIdentifier attribute or generate a version 1 UUID if there is no value set for the attribute
 	 */
 	@Override
 	public Attribute fillAttribute(PerunSessionImpl perunSession, User user, AttributeDefinition attribute) {
 
 		Attribute filledAttribute = new Attribute(attribute);
 
 		if (attribute.getFriendlyName().equals(FRIENDLY_NAME)) {
-			long most64SigBits = get64MostSignificantBits();
-			long least64SigBits = get64LeastSignificantBits();
-			UUID uuid = new UUID(most64SigBits, least64SigBits);
 
-			filledAttribute.setValue(uuid.toString());
+			Attribute umbrellaIDIdentifier = null;
+
+			try {
+				umbrellaIDIdentifier  = perunSession.getPerunBl().getAttributesManagerBl().getAttribute(perunSession, user, A_U_UMBRELLAID_IDENTIFIER);
+			} catch (WrongAttributeAssignmentException e) {
+				throw new InternalErrorException(e);
+			} catch (AttributeNotExistsException e) {
+				//this is ok, attribute will be generated.
```

Comment:
I would at least ``log.warn()``, since it means, that attribute definition of sourcing attribute value doesn't exists. It is not mandatory, but otherwise you might not notice, that attribute is missing at all.

Reply:
Fixed